lib/puppet/context.rb

```diff
@@ -40,6 +40,7 @@ def pop
     if @stack[-1][0] == 0
       raise(StackUnderflow, "Attempted to pop, but already at root of the context stack.")
     else
+      @id -= 1
```

Comment:
I'm pretty sure `@id` is meant to be a unique id for things in the stack, not the stack depth. Although it isn't exposed outside of this class, I'm thinking we shouldn't modify it's behavior, especially since it's not meaningful to this PR.